client/src/redux/Form/formReducer.ts

```diff
@@ -0,0 +1,18 @@
+import * as Actions from './formActionTypes';
+
+const initialState : any = {}; //[null,null,null,null]
+
+const formReducer = (state = initialState, action: Actions.formAction) : (boolean | null)[] => {
+    switch (action.type) {
+        case Actions.SET_FORM_STATE : {
+            if(state[action.payload.investigationId] === undefined) {
+                state[action.payload.investigationId] = [null,null,null,null];
```

Comment:
why don't initiate with false?

Reply:
null is for unvisited tab and false is for invalid visited tab -